beacon-chain/cache/depositcache/deposits_cache.go

```diff
@@ -162,7 +162,10 @@ func (dc *DepositCache) DepositByPubkey(ctx context.Context, pubKey []byte) (*et
 
 	var deposit *ethpb.Deposit
 	var blockNum *big.Int
-	for _, ctnr := range dc.deposits {
+	// Search the list backwards since this function is only used for
+	// pending activation validators and the newest deposits are always at the end.
+	for i := len(dc.deposits) - 1; i > 0; i-- {
```

Comment:
If anyone disagrees with this change please let me know, this function is only used for newer validators so I figured it'd make sense.

Reply:
This makes sense, nice catch!

Reply:
Please change this back, this picks the wrong deposit in the event of top-ups. Its why the tests fail now.